resolver/resolver.go

```diff
@@ -286,6 +286,11 @@ func (t Target) Endpoint() string {
 	return strings.TrimPrefix(endpoint, "/")
 }
 
+// String returns a string representation of Target.
+func (t Target) String() string {
+	return t.URL.String()
+}
+
```

Comment:
These changes collide with the other pr you sent out, just a heads up. This and scheme.

Reply:
I've rebased now and cleanup things a bit. Will merge once GitHub Actions is happy.